be/src/storage/rowset/segment_v2/column_reader.cpp

```diff
@@ -189,6 +189,9 @@ Status ColumnReader::read_page(const ColumnIteratorOptions& iter_opts, const Pag
     opts.verify_checksum = _opts.verify_checksum;
     opts.use_page_cache = iter_opts.use_page_cache;
     opts.kept_in_memory = _opts.kept_in_memory;
+    if (is_data_page) {
```

Comment:
why should add is_data_page here?

Reply:
> 

because `column_reader`  will be used  to read `data_page`, `dict_page` and other types of page, but only `data_page` should be decompressed in advance